src/richie/apps/courses/models/course.py

```diff
@@ -330,9 +330,12 @@ def compute_state(start, end, enrollment_start, enrollment_end):
     @property
     def state(self):
         """Return the state of the course run at the current time."""
-        return self.compute_state(
-            self.start, self.end, self.enrollment_start, self.enrollment_end
-        )
+        if self.start:
```

Comment:
I think you need to do this inside the `compute_state` method instead of here otherwise the `compute_state` method is still broken.

Reply:
The problem is that the linter complains that there are too many return statement. Not sure how to get around this problem.

Reply:
Right so just insert the state into a value and return it at the end. Got it.